### DISCUSSION TITLE
Floppy image read error

### DISCUSSION THREAD

**Discussion by oilcan-productions** [thehighnibble/thehighnibble.github.io#89] [Q&A]
I used cpmtools to create a floppy image on Ubuntu

`mkfs.cpm blank.dsk`

then used `cpmcp blang.dsk *.* 0:` to copy some files to the disk

I uploaded the dsk to the IMSAI8080 using the web interface
Mounted the floppy image in drive B:
when accessing the drive i.e. to do a `DIR` I get this error `Bdos Err On B: Bad Sector`

comparing the disk to another one downloaded from the library with `fsed.cpm` shows no difference. They both seem to be formatted the same way. 

Any pointers on what could be wrong?

**Comment by oilcan-productions** [comment]
Found a awkward work around:
1. Create DSK image file with `mkfs,cpm`
2. import the DSK file into the library
3. mount the DSK file into an empty drive i.e. `B:`
4. Switch to `A;`
5. run `SYSGEN` specify the new image as the destination drive
6. Type `ERA B:*.*`
7. Download the image file
8. Now you have an empty diskimage you can the cpmtools to copy other files onto

**Reply by pfcushing** [reply]
did you specify the disk format using the option "-f ibm-3740" in the mkfs.cpm operation?

**Reply by oilcan-productions** [reply]
It is the default format. I tried specifying it as well no change

**Comment by dmcnaugh** [comment]
I just followed your steps and observed that `mkfs.cpm` **isn't** creating a file that is the full size of a disk image (256,256 bytes)
It is only writing enough data to the file to account for the boot tracks (1&2) and the directory (tacks 3&4) or there-abouts (9,984 bytes)

The undersized file is likely the problem. Z80PACK expects a full sized disk image.

When you perform the `sysgen` under CP/M (I haven't checked this myself, yet) I assume the file is getting padded out to the full disk image size.

I always start with a copy of an existing disk image and just delete everything from it before declaring it "blank".
This isn't secure as CP/M (and presumably `cpmtools`) doesn't delete anything, it just marks a directory extent as unused by setting the `user` field to `0xE5`.

To create a truly blank floppy disk image you need to generate a file that contains exactly 256,256 bytes all with the value 0xE5
